### ISSUE TITLE
Cannot compile libnih on Arch Linux

### ISSUE THREAD

**Issue by luca3m** [nezticle/RaspberryPi-BuildRoot#34]
Hi

I'm trying to create a raspberry image on Arch Linux.

I get an error while compiling libnih, it doesn't find dbus libraries trough pkg-config. So I tried to set DBUS_CFLAGS and DBUS_LIBS in this way:

``` sh
➜  build git:(master) ✗ echo $DBUS_CFLAGS
-I/usr/include/dbus-1.0 -I/usr/lib/dbus-1.0/include
➜  build git:(master) ✗ echo $DBUS_LIBS
-ldbus-1
➜  build git:(master) ✗
```

It goes well to find includes but I get an error on linking:

``` sh
property.c:(.text+0x369): undefined reference to `dbus_error_init'
property.c:(.text+0x379): undefined reference to `dbus_signature_validate_single'
property.c:(.text+0x5ff): undefined reference to `dbus_error_free'
property.o: In function `property_object_get_function':
property.c:(.text+0xf01): undefined reference to `dbus_signature_iter_init'
property.o: In function `property_object_set_function':
property.c:(.text+0x174b): undefined reference to `dbus_signature_iter_init'
property.o: In function `property_proxy_get_notify_function':
property.c:(.text+0x286a): undefined reference to `dbus_signature_iter_init'
property.o: In function `property_proxy_set_function':
property.c:(.text+0x33af): undefined reference to `dbus_signature_iter_init'
property.o: In function `property_proxy_set_notify_function':
property.c:(.text+0x3f2d): undefined reference to `dbus_signature_iter_init'
property.o:property.c:(.text+0x467e): more undefined references to `dbus_signature_iter_init' follow
argument.o: In function `argument_start_tag':
argument.c:(.text+0x385): undefined reference to `dbus_error_init'
argument.c:(.text+0x395): undefined reference to `dbus_signature_validate_single'
argument.c:(.text+0x73e): undefined reference to `dbus_error_free'
```

Do you know how to solve this?

**Comment by nezticle** [comment]
Are you building from a fresh build?  In the last update I had to bump of the version of dbus that gets build because of libnih and upstart.  the errors you are getting are likely a result of either using the older dbus version from a previous build, or your system dbus version (which may be older) getting used instead.  I test builds from a Ubuntu 12.04 installation, but no generally only system tools like flex and bison, and not development libs/includes.

**Comment by qdotme** [comment]
I can confirm the bug (failure to build dbus) on fresh Gentoo AMD64 running fresh build of bsquask.

Gentoo and Arch are reasonably similar in design - so something might be causing this issue. Please let me know how to assist you in debugging it further. 

```
checking for DBUS... no
configure: error: Package requirements (dbus-1 >= 1.2.16) were not met:

Package dbus-1 was not found in the pkg-config search path.
Perhaps you should add the directory containing `dbus-1.pc'
to the PKG_CONFIG_PATH environment variable
No package 'dbus-1' found

Consider adjusting the PKG_CONFIG_PATH environment variable if you
installed software in a non-standard prefix.

Alternatively, you may set the environment variables DBUS_CFLAGS
and DBUS_LIBS to avoid the need to call pkg-config.
See the pkg-config man page for more details.
make: *** [/opt/bsquask/build/host-libnih-1.0.3/.stamp_configured] Error 1
```

The exact same build (git clone out of local copy) builds just fine on a fresh Ubuntu 12.04.2 Server.

**Comment by wibbler** [comment]
Get the same problem with Slackware 14.0 (x86).....

**Comment by nezticle** [comment]
ok, I didn't realize this was an error with libnih-host (the one built for the host machine linked against your local dbus).  It's not a problem for Bsquask, because I can upgrade dbus to whatever version I need, but less than ideal that I depend on the build host to have a certain version.

I use upstart to get super-fast boot times, but this requires libnih, which requires a very recent release of dbus.  For some reason (that I dont recall), I need to build a host version of libnih, which also needs a recent release of dbus.  It's not a problem for people using Ubuntu, because they obviously have the same upstart dependencies locally, but for everyone else it could be an issue.

I will investigate this further.

**Comment by nezticle** [comment]
Ok, so the reason why I have to build a host version of libnih is for a cool called nih-dbus-tool, which generates C bindings for dbus objects.  When I'm cross compiling Upstart for bsquask, this nih-dbus-tool is needed as a build tool, and it needs to be runnable from the host machine.

The tool links against libdbus-1.so.3, and actually requires API's only available in libdbus greater than 1.2.16

Normally I would just build a host version of dbus as well, but dbus is not a simple library, but rather a library interface to a system daemon.  It may be possible though to build just libdbus, since we wont actually be interacting with the system dbus IPC daemon.

So investigating a solution...

**Comment by nezticle** [comment]
I double-checked my own installation, and libdbus-1.so.3 is provided by the host-dbus package in bsquask.  That means that it is possible to build libdbus as a host lib.  This wasn't set an an explicit dependency of host-libnih, so there was no guarantee that it would be in place before we try and build libnih.

This "should" be fixed by: cca05d35e9d663b15e15ea691125f9f94ac1c389

If not, re-open this task and we shall try again.
